source/common/quic/envoy_quic_client_session.cc

```diff
@@ -64,6 +64,9 @@ EnvoyQuicClientSession::EnvoyQuicClientSession(
       quic_stat_names_(quic_stat_names), rtt_cache_(rtt_cache), scope_(scope),
       transport_socket_options_(transport_socket_options) {
   streamInfo().setUpstreamInfo(std::make_shared<StreamInfo::UpstreamInfoImpl>());
+  // HTTP/3 Datagrams sent over CONNECT-UDP are already congestion controlled, so makes it bypass
+  // the default Datagram queue.
+  SetForceFlushForDefaultQueue(true);
```

Comment:
Is this universal enough that we don't need to check ALPN?

Reply:
That queue is only for HTTP/3 Datagrams so this option is relevant to CONNECT-UDP only.

Reply:
My reading of the code indicates that it is used by all MESSAGE frame. I assume it's the super set of CONNECT-UDP?

I understand that we want to disable congestion control for tunneling case. But I'm wondering if there exists a usage of MESSAGE frame where we want connection level congestion control in the future.

Reply:
Right, I meant that in Envoy now, CONNECT-UDP is the first and only use case that uses the queue. In the future there maybe a case that requires congestion control on the proxy side.

I will change where `SetForceFlushForDefaultQueue` is called in a subsequent commit, so that it's called only when CONNECT-UDP is used.